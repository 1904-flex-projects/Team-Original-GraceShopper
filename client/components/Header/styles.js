export const HEADER_CONTAINER = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  backgroundColor: '#0046be',
  width: '100vw',
  color: 'white',
  fontFamily:
    'Human BBY Web,Arial,Helvetica,Nimbus Sans L,Liberation Sans,FreeSans,sans-serif',
  textDecoration: 'none',
};

export const NAV_LINKS_CONTAINER = {
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  width: '100vw',
  margin: '10px 0px 10px 0px',
};

export const NAV_ACCOUNT_LINKS_CONTAINER = {
  display: 'flex',
  flexDirection: 'row',
};

export const NAV_LINK = {
  paddingRight: '20px',
};

export const PRODUCT_MODAL = {
  paddingLeft: '20px',
};

export const SEARCH_INPUT = {
  width: '50vw',
};

export const SEARCH_BAR = {
  width: '100vw',
};

export const ACCOUNT_PROFILE = {
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  margin: '40px 100px 0 100px',
  padding: '20px',
};

export const COMPANY_NAME = {
  textDecoration: 'none',
  color: 'white',
};

export const NAV_IMAGE = {
  width: '12px',
};
