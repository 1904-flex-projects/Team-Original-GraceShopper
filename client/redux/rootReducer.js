import { combineReducers } from 'redux';

// import individual reducers here
import currentUserReducer from './reducers/currentUserReducer';
import isLoggedInReducer from './reducers/isLoggedInReducer';
import productReducer from './reducers/productReducer';

const rootReducer = combineReducers({
  isLoggedIn: isLoggedInReducer,
  currentUser: currentUserReducer,
  product: productReducer,
});

export default rootReducer;
