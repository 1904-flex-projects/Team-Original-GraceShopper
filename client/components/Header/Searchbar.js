import React from 'react';
import { SEARCH_INPUT } from './styles';

const Searchbar = () => {
  return (
    <div
      id="search-bar"
      style={{
        width: '100vw',
      }}
    >
      <div />
      <form action="submit">
        <div style={{ display: 'flex', justifyContent: 'center' }}>
          <input
            type="text"
            name="search"
            placeholder="search"
            style={SEARCH_INPUT}
          />
          <button type="submit"> Search</button>
        </div>
        <hr style={{ width: '100vw' }} />
      </form>
    </div>
  );
};

export default Searchbar;
