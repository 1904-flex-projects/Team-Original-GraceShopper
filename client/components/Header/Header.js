import React from 'react';

import Navbar from './Navbar/Navbar';
import Searchbar from './Searchbar';

import { HEADER_CONTAINER } from './styles';

const Header = () => {
  return (
    <header id="header-container" style={HEADER_CONTAINER}>
      <h2>Bester Buy</h2>
      <Searchbar />
      <Navbar />
    </header>
  );
};

export default Header;
