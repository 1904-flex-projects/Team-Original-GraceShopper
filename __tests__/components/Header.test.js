import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Header from './../../client/components/Header/Header';

xtest('it renders correctly', () => {
  const reducer = () => {
    return {};
  };
  const store = createStore(reducer);
  const component = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
