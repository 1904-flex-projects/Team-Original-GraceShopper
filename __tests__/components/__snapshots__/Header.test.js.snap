// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<header
  id="header-container"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0046be",
      "color": "white",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "Human BBY Web,Arial,Helvetica,Nimbus Sans L,Liberation Sans,FreeSans,sans-serif",
      "textDecoration": "none",
    }
  }
>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={
      Object {
        "color": "white",
        "textDecoration": "none",
      }
    }
  >
    <h1>
      Bester Buy
    </h1>
  </a>
  <div
    id="search-bar"
    style={
      Object {
        "width": "100vw",
      }
    }
  >
    <form
      action="submit"
    >
      <div
        style={
          Object {
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <input
          name="search"
          placeholder="Search"
          style={
            Object {
              "width": "50vw",
            }
          }
          type="text"
        />
        <button
          type="submit"
        >
           Search
        </button>
      </div>
      <hr />
    </form>
  </div>
  <div
    id="nav-links-container"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": "10px 0px 10px 0px",
        "width": "100vw",
      }
    }
  >
    <div
      id="nav-products-container"
    >
      <a
        aria-current={null}
        href="/products"
        onClick={[Function]}
        style={
          Object {
            "paddingLeft": "20px",
          }
        }
      >
        <button>
          Products
        </button>
      </a>
      <a
        aria-current={null}
        href="/brands"
        onClick={[Function]}
        style={
          Object {
            "paddingLeft": "20px",
          }
        }
      >
        <button>
          Brands
        </button>
      </a>
    </div>
    <div
      id="nav-account-links-container"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      
      <a
        aria-current={null}
        href="/login"
        onClick={[Function]}
        style={
          Object {
            "paddingRight": "20px",
          }
        }
      >
        <button>
          Login
        </button>
      </a>
      <a
        aria-current={null}
        href="/account"
        onClick={[Function]}
        style={
          Object {
            "paddingRight": "20px",
          }
        }
      >
        <button>
          A
          ccount
        </button>
      </a>
      <a
        aria-current={null}
        href="/orders"
        onClick={[Function]}
        style={
          Object {
            "paddingRight": "20px",
          }
        }
      >
        <button>
          O
          rders
        </button>
      </a>
      <a
        aria-current={null}
        href="/cart"
        onClick={[Function]}
        style={
          Object {
            "paddingRight": "20px",
          }
        }
      >
        <button>
          <img
            src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcR8Bncj1OxlP5IoFnjojgmEwkmlOUYGFRe4cxmvHtihDVXt--SURA"
            style={
              Object {
                "width": "12px",
              }
            }
          />
           
          C
          art
        </button>
      </a>
    </div>
  </div>
</header>
`;
